src/Executor.cpp

```diff
@@ -278,51 +278,69 @@ void ExecutorADDI(Hart* hart, const DecodedInstruction& instr) {
 void ExecutorSLLI(Hart* hart, const DecodedInstruction& instr) {
     DEBUG_INSTRUCTION("slli    x%d, x%d, %d\n", instr.rd, instr.rs1, instr.shamt);
 
+    uint64_t res = hart->getReg(instr.rs1) << (instr.shamt);
+    hart->setReg(instr.rd, res);
     hart->incrementPC();
 }
 
 void ExecutorSLTI(Hart* hart, const DecodedInstruction& instr) {
     DEBUG_INSTRUCTION("slti    x%d, x%d, x%d\n", instr.rd, instr.rs1, instr.rs2);
 
+    uint64_t res = hart->getReg(instr.rs1) < instr.imm;
+    hart->setReg(instr.rd, res);
     hart->incrementPC();
 }
 
 void ExecutorSLTIU(Hart* hart, const DecodedInstruction& instr) {
     DEBUG_INSTRUCTION("sltiu   x%d, x%d, x%d\n", instr.rd, instr.rs1, instr.rs2);
 
+    auto left = hart->getReg(instr.rs1);
```

Comment:
Use explicit types instead of auto here. You need exactly `uint64_t`. And in SLTI you need do the same but `int64_t`